src/component/AddItem.js

```diff
@@ -3,27 +3,49 @@ import { withFirestore } from "react-firestore";
 import "../App.css";
 import Navbar from "./Navbar";
 import BackButton from "./BackButton";
+import DuplicateMessage from "./DuplicateMessage";
 
 const AddItem = ({ firestore }) => {
   const [name, setName] = useState("");
-  const [frequency, setFrequency] = useState("14");
+  const [numberOfDays, setNumberOfDays] = useState("14");
+  const [duplicate, setDuplicate] = useState(false);
+
   const uniqueToken = localStorage.getItem("uniqueToken");
 
   //   Write item to Firebase setting uniqueToken as document name
-  const addItem = (name, frequency) => {
+  const addItem = (normalizedName, numberOfDays) => {
+    // adds new items collection to database
     firestore
       .collection("lists")
       .doc(uniqueToken)
       .set({ items: "" });
 
-    firestore
+    // reference path to specific document from items collection
+    // sets document ID equal to item name
+    const itemsDocRef = firestore
       .collection("lists")
       .doc(uniqueToken)
       .collection("items")
-      .add({
-        name: name,
-        frequency: +frequency
-      });
+      .doc(normalizedName);
+
+    // checks whether an existing doc ID is equal to new item name
+    itemsDocRef.get().then(docSnapshot => {
+      if (docSnapshot.exists) {
+        let timeWindowBeforeRefresh = 2500;
```

Comment:
That's what I meant, thanks for changing this!

Ideally you'd want this to live outside of this `then` callback function because you're initializing it every time the callback executes.

If you decide to move it, I'd suggest moving it outside of the component and use `const` instead of `let` because it is a constant :)